dds/DCPS/MultiTopicDataReaderBase.cpp

```diff
@@ -134,17 +149,25 @@ void MultiTopicDataReaderBase::init(const DDS::DataReaderQos& dr_qos,
 
     QueryPlan& qp = query_plans_[selection[i]];
     qp.data_reader_ = incoming;
-    const MetaStruct& meta = metaStructFor(incoming);
-
-    for (const char** names = meta.getFieldNames(); *names; ++names) {
-      if (fieldToTopic.count(*names)) { // already seen this field name
-        set<OPENDDS_STRING>& topics = joinKeys[*names];
-        topics.insert(fieldToTopic[*names]);
-        topics.insert(selection[i]);
-      } else {
-        fieldToTopic[*names] = selection[i];
+    try {
+      const MetaStruct& meta = metaStructFor(incoming);
+
+      for (const char** names = meta.getFieldNames(); *names; ++names) {
+        if (fieldToTopic.count(*names)) { // already seen this field name
+          set<OPENDDS_STRING>& topics = joinKeys[*names];
+          topics.insert(fieldToTopic[*names]);
+          topics.insert(selection[i]);
+        }
+        else {
```

Comment:
Please check formatting of the changed lines so they match the style guidelines.